cmd/junitreport/pkg/parser/interfaces.go

```diff
@@ -0,0 +1,61 @@
+package parser
+
+import (
+	"bufio"
+
+	"github.com/openshift/origin/cmd/junitreport/pkg/api"
+)
+
+// TestOutputParser knows how to parse test output to create a collection of test suites
+type TestOutputParser interface {
+	Parse(input *bufio.Scanner) (*api.TestSuites, error)
+}
+
+// TestDataParser knows how to take raw test data and either characterize it as data that
+// contains useful information or extract the useful information from it
+type TestDataParser interface {
```

Comment:
If we want a single interface for extraction we have already agreed that we need `ExtractIdentifer`, `ExtractResult`, and not `identifier, result, whatever, time := ParseGoTestResultLine`. Therefore, we're already multiplying the number of regex calls we are doing. I am not a huge fan of methods that return two values, one of which you need to check to see if the other is useful... so I don't really mind doubling the work again. I agree the current implementation of "figure out if it goes, then stick it in something and extract later" isn't the best, and I'm about to refactor it, but I am not convinced that removing the `ContainsX` methods is a good idea for readability.